app/src/main/java/org/mozilla/rocket/content/travel/data/BcApiEntity.kt

```diff
@@ -100,22 +101,30 @@ data class BcHotelApiItem(
         private const val KEY_ROOM_INFO = "room_info"
         private const val KEY_ROOM_INFO_MIN_PRICE = "min_price"
 
-        private const val FACILITY_FREE_WIFI = "free_wifi_internet_access_included"
-        private const val PAY_AT_PROPERTY = "1"
+        private const val FACILITY_ID_FREE_WIFI = "107"
 
         fun fromJson(jsonObject: JSONObject): BcHotelApiItem? {
 
             try {
                 val hotelData = jsonObject.optJSONObject(KEY_HOTEL_DATA)
                 val hotelPhotos = hotelData.optJSONArray(KEY_DATA_HOTEL_PHOTOS)
-                val hotelPhotoItem = hotelPhotos.getJSONObject(0)
                 val hotelFacilities = hotelData.optJSONArray(KEY_DATA_HOTEL_FACILITIES)
 
+                var hotelPhotoItem: JSONObject? = null
+                for (i in 0 until hotelPhotos.length()) {
+                    val photoItem = hotelPhotos.getJSONObject(i)
+                    if (photoItem.optBoolean(KEY_DATA_HOTEL_PHOTOS_MAIN, false)) {
+                        hotelPhotoItem = photoItem
+                        break
+                    }
+                }
+                hotelPhotoItem = hotelPhotoItem ?: hotelPhotos.getJSONObject(0)
```

Comment:
1. It could define separate `hotelMainPhotoItem` so you can have a non-nullable `hotelPhotoItem` then `!!` can be removed from  `hotelPhotoItem!!.optString(KEY_DATA_HOTEL_PHOTOS_URL_ORIGINAL`
2. If the photo list is empty, the fallback `hotelPhotos.getJSONObject(0)` would be null. Need to add error handling for that. Maybe use another way to get the image url.

Reply:
```
val imageUrl = if (hotelPhotos.length() > 0) {
                    var hotelMainPhotoItem: JSONObject? = null 
                    for (i in 0 until hotelPhotos.length()) {
                        val photoItem = hotelPhotos.getJSONObject(i)
                        if (photoItem.optBoolean(KEY_DATA_HOTEL_PHOTOS_MAIN, false)) {
                            hotelMainPhotoItem = photoItem
                            break
                        }
                    }
                    val hotelPhotoItem = hotelMainPhotoItem ?: hotelPhotos.getJSONObject(0)
                    hotelPhotoItem.optString(KEY_DATA_HOTEL_PHOTOS_URL_ORIGINAL)
                } else {
                    ""
                }
```